src/main/java/org/spongepowered/api/event/player/PlayerRespawnEvent.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.player;
+
+import org.spongepowered.api.world.Location;
+
+/**
+ * Called when a player respawns after death.
+ */
+public interface PlayerRespawnEvent extends PlayerEvent {
+
+    /**
+     * Gets the respawn location of the player.
+     *
+     * @return The respawn location of the player
+     */
+    Location getRespawnLocation();
+
+    /**
+     * Gets whether the respawn location was set by a bed or not.
+     *
+     * @return Whether the respawn location was set by a bed
+     */
+    boolean isBedSpawn();
+
+    /**
+     * Sets the new player respawn location permanently.
+     *
+     * @param respawnLocation The new respawn location
+     */
+    void setSpawnLocation(Location respawnLocation);
```

Comment:
This comment by @gabizou was written on the leavebedevent but it makes more sense as a reply here.

    Sometimes a one time respawn change is needed without changing the more persistent respawn location.

    If the persisting respawn point is to be set, Player#setRespawnPoint should work.